### ISSUE TITLE
behavior of cifti-replace-structure seems altered in v1.5 -- leading to new errors

### ISSUE THREAD

**Issue by ericfeczko** [Washington-University/workbench#52]
Hey all,

I'm noticing that the behavior for a low-level command is different. 

I have a series of separated ROI files that I need to add to a dtseries, replacing the contents of each subcortical structure at a time. In 1.4.2, this worked just fine by creating a temporary dtseries comprising all the data, and then replacing the structure of each dtseries component at a time:

`wb_command -cifti-create-dense-timeseries ses-24mo_task-rest_run-01_temp_orig_atlas.dtseries.nii -volume ses-24mo_task-rest_run-01_2MNI.nii.gz ROIs/Atlas_ROIs.2.nii.gz`

`wb_command -cifti-create-dense-from-template ses-24mo_task-rest_run-01_temp_orig_atlas.dtseries.nii ses-24mo_task-rest_run-01_temp_atlas.dtseries.nii -series 0.8 0.0 -volume CEREBELLUM_LEFT ses-24mo_task-rest_run-01_ROI0000.nii.gz`

`wb_command -cifti-replace-structure ses-24mo_task-rest_run-01_temp_atlas.dtseries.nii COLUMN -volume THALAMUS_LEFT ses-24mo_task-rest_run-01_ROI0001.nii.gz`

In 1.5.0, such a strategy produces an error:

"While running:
/panfs/roc/groups/8/faird/shared/code/external/utilities/workbench/1.5.0/workbench/bin_linux64/../exe_linux64/wb_command -cifti-replace-structure ses-24mo_task-rest_run-01_temp_atlas.dtseries.nii COLUMN -volume THALAMUS_LEFT ses-24mo_task-rest_run-01_ROI0001.nii.gz

ERROR: -volume specified more than once with structure THALAMUS_LEFT"

I'm looking for an alternative way of doing this, if anyone has a workaround, that would be great.

**Comment by coalsont** [comment]
This looks like an added sanity check was written incorrectly.  You could work around it by using 1.4.2 for this command, or by using the -volume argument multiple times in the single -cifti-create-dense-from-template command (it is a repeatable option), which will also do less disk IO, though it will have more memory usage as it opens all the input files at once.  If -cifti-separate is what you used to separate all the volume structures to begin with, you can use the -crop and -from-cropped options to reduce the FoV of each volume file to reduce the memory usage.  I don't know if what you are doing on each structure would be amenable to working with the -volume-all output and a volume label file of the structure boundaries.

The dev_latest version here should fix the issue, if you don't mind trying it to make sure the next release doesn't have this issue:

http://brainvis.wustl.edu/workbench/

The dev_latest version will have some unfinished features, please don't make scenes using its wb_view, as there could be incompatible changes to the new features before the next release.

As for what you are doing, that -cifti-create-dense-timeseries command will make a voxels-only cifti file, which could be inconvenient to compare to standard 91k grayordinates files.

**Comment by ericfeczko** [comment]
Oh, yeah we want a voxels-only cifti file for what we're using here, eventually we re-extract the volumetric component back out later in the pipeline. 

In fact, the problem occurred because we were looking for a way to reduce RAM usage, and didn't mind the additional file IO ( we are merging more than 20 files, which creates a huge RAM crunch on our systems ). I switched from the -cifti-create-dense-from-template to -cifti-replace-structure for that very reason 😄  good to know that I can try using the -crop and -from-cropped to keep the RAM usage lower, thanks! 

I did find another workaround -- since the volumetric files are all non-overlapping, I had done what you suggested and added them, then used a volume-label file to make the dtseries via -cifti-create-dense-timeseries. The files are all indexed already to subcortical structures within the CIFTI annotation, so that worked to keep the ram usage low. 

Sure, I'd be happy to try it, and see if that fixes the issue :)
